### ISSUE TITLE
How to use aggregate with typed collections?

### ISSUE THREAD

**Issue by mityax** [PoloDB/PoloDB#162]
I can't use `aggregate()` with a collection that has a struct defined as it's type:

```rust
let collection = self.db.collection<MyStruct>("my_structs");
let count = collection
    .aggregate(pipeline)
    .run()
    .expect("Could not execute database aggregate")
    .collect::<polodb_core::Result<Vec<Document>>>()
    .unwrap()[0].get("count").unwrap().as_i64().unwrap()
```

When trying to compile, I get:

```
error[E0277]: a value of type `Vec<polodb_core::bson::Document>` cannot be built from an iterator over elements of type `MyStruct`
    --> src/my_file.rs:60:24
     |
60   |             .collect::<polodb_core::Result<Vec<Document>>>()
     |              -------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ value of type `Vec<polodb_core::bson::Document>` cannot be built from `std::iter::Iterator<Item=MyStruct>`
     |              |
     |              required by a bound introduced by this call
     |
```

What am I doing wrong here/how can I fix that?

**Comment by vincentdchan** [comment]
Because your collection is defined as `self.db.collection<MyStruct>`. So you can't collection as Document.
So correct the collection Document should fix this
```rust
let collection = self.db.collection<Document>("my_structs");
```
This should be considered as an issue. `aggregate` should return Document by default.
Will fix in next version

**Comment by vincentdchan** [comment]
Fixed in `v5.1.1`
